hanadb_exporter/db_manager.py

```diff
@@ -139,3 +145,42 @@ def get_connectors(self):
         Get the connectors
         """
         return self._db_connectors
+
+    def get_active_master_host(self, hosts, master_host, master_port):
```

Comment:
As far as I see, here you are looping among all the nodes in the scale out system to find out where the SYSTEM db is running. Did you check if this information is available in the database itself, so we can space all this looping and checking?